request/requestid.go

```diff
@@ -0,0 +1,20 @@
+package request
+
+import (
+	"crypto/rand"
+	"fmt"
+
+	"github.com/oklog/ulid"
+)
+
+// ID is an identifier to uniquely identify a request
+type ID string
```

Comment:
The Request ID is in the form `<timestamp>-<random_id>` and we use the timestamp part, for example, store request information partitioned by time. Instead of a string, what do you think about having it be a struct containing these two parts?